src/storage/core.rs

```diff
@@ -795,12 +795,24 @@ impl<K: Key> Storage<K> {
             .read()
             .await
             .iter()
-            .map(|blob| blob.check_filters(key.as_ref()))
-            .collect::<FuturesUnordered<_>>()
-            .any(|value| value)
-            .await;
+            .filter(|blob| !blob.is_filter_offloaded())
+            .any(|blob| blob.check_filters_in_memory(key.as_ref()));
 
-        Some(in_active || in_closed)
+        if !(in_active || in_closed) {
```

Comment:
I think the code will become more readable if you write it like this: 
```
let in_active = ...;
if (in_active)
    return Some(true);
let in_closed = ...;
if (in_closed)
    return Some(true);
let in_closed_offloaded = ...;
return Some(in_closed_offloaded)
```

Reply:
Done